Assets/MRTK/Providers/WindowsMixedReality/Shared/InputHandlers/HPMotionControllerInputHandler.cs

```diff
@@ -0,0 +1,246 @@
+                            break;
+                        case DeviceInputType.Menu:
+                            button = ControllerInput.Menu;
+                            break;
+                        case DeviceInputType.ThumbStickTouch:
+                        case DeviceInputType.ThumbStickPress:
+                            button = ControllerInput.Thumbstick;
+                            break;
+                        default:
+                            return;
+                    }
+
+
+                    var buttonData = controllerState.CurrentReading.GetPressedValue(button);
+                    interactionMapping.BoolData = buttonData > 0.0f;
+                }
+
+                // If our value changed raise it.
+                if (interactionMapping.Changed)
+                {
+                    // Raise input system event if it's enabled
+                    if (interactionMapping.BoolData)
+                    {
+                        CoreServices.InputSystem?.RaiseOnInputDown(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
+                    }
+                    else
+                    {
+                        CoreServices.InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
+                    }
+                }
+            }
+        }
+
+        private static readonly ProfilerMarker UpdateSingleAxisDataPerfMarker = new ProfilerMarker("[MRTK] HPMotionControllerInputHander.UpdateSingleAxisData");
+
+        /// <summary>
+        /// Update an interaction float data type from a SingleAxis (float) input
+        /// </summary>
+        /// <remarks>
+        /// Raises a FloatInputChanged event when the float data changes
+        /// </remarks>
+        internal virtual void UpdateSingleAxisData(MixedRealityInteractionMapping interactionMapping, MotionControllerState controllerState)
+        {
+            using (UpdateSingleAxisDataPerfMarker.Auto())
+            {
+                Debug.Assert(interactionMapping.AxisType == AxisType.SingleAxis);
+                // Update the interaction data source
+                switch (interactionMapping.InputType)
+                {
+                    case DeviceInputType.Trigger:
+                        var triggerData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Trigger);
+                        interactionMapping.BoolData = !Mathf.Approximately(triggerData, 0.0f);
+                        interactionMapping.FloatData = triggerData;
+                        break;
+                    case DeviceInputType.Grip:
+                        var gripData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Grasp);
+                        interactionMapping.BoolData = !Mathf.Approximately(gripData, 0.0f);
```

Comment:
These feel like the `GripTouch` and `TriggerTouch` cases. It almost feels like these should be removed from here, placed up in the lines 102-111 if statements (instead of the check against `.Equals(1)`), then  `GripTouch` and `TriggerTouch` in the line 117 switch statement get swapped with `TriggerPress` and `GripPress` from those previously mentioned if statements. Does that seem right, or am I missing something?

Reply:
Yeah I think I made a mistake with raising trigger/grip InputDown and InputUp events here. This is the final break down based on my understanding.

UpdateButton:
Raise Touch/NearTouch events when float > 0.0f, raise Press/GripPress events when float == 1.0f

Update Single Axis 
raise TriggerPress/GripPress events when float == 1.0f (based on the GenericJoystickController implementation, check is done here as well as Update Button for comprehensiveness)

Raise Trigger/Grip float change events whenever the float changes.

Reply:
>raise Press/GripPress events when float == 1.0f

I think the button, menu, thumbstick press events etc should fall in this case too (but looks like right now they fall in the `Touch` cases). Maybe the special-cased `if`s before the `switch` become the `Touch` checks, and the Press events can move to the `switch` with the rest of the button-based events?